### PR TITLE
inventory aws_ec2 - allow for literal strings in hostnames

### LINKED ISSUE #814: Composing inventory_hostname manually
### Summary

Here is my current config:
```yaml
---
plugin: amazon.aws.aws_ec2
regions:
  - eu-central-1
hostnames:
  - tag:Name
compose:
  ansible_host: private_ip_address
```

Executing it sets inventory_hostname variable(s) to the "Name" tag value. However, I would like to know whether it's possible to create a completely custom inventory_hostname, as I haven't found a clear way to do it by searching through the docs (as far as I understood it's only allowed to use the values from [here](https://docs.aws.amazon.com/cli/latest/reference/ec2/describe-instances.html#options) which doesn't suit my case).

I have an AWS cloud and an Azure cloud, which serves as a mirror of the AWS cloud. The VMs on both clouds are named similarly (e.g. "prometheus"). I would like to refer to these VMs as 'aws-prometheus' and 'azu-prometheus'. The [azure_rm plugin](https://docs.ansible.com/ansible/latest/collections/azure/azcollection/azure_rm_inventory.html#parameter-hostnames) allows creating custom inventory_hostname:

```yaml
hostnames:
  - "'azu-' + name"
```
, where "name" is just a key in the object returned by the API (like 'private_ip_address' for AWS). This way, I can refer to Azure VMs with 'azu-' prefix.

I would like to know whether it's possible to do the same with the aws_ec2 plugin. I've played around with different combinations and values for 'hostnames' but found myself limited with the values allowed by AWS.

This doesn't work because "aws-" is not a recognized value from [here](https://docs.aws.amazon.com/cli/latest/reference/ec2/describe-instances.html#options):
```yaml
---
plugin: amazon.aws.aws_ec2
regions:
  - eu-central-1
hostnames:
  - name: "tag:Name"
    prefix: "aws-"
compose:
  ansible_host: private_ip_address
```

Thank you in advance.

### Issue Type

Feature Idea

### Component Name

amazon.aws.aws_ec2

### Additional Information

_No response_

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by jatorcasso**
##### SUMMARY
Allow for literal strings in hostnames rather than throwing an AnsibleError

Fixes #814

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
aws_ec2

##### ADDITIONAL INFORMATION
Is there a better way of allowing for literal strings here?

### CODE REVIEW

**Review by markuman — COMMENTED** [review]
It would be great if the `DOCUMENTATION` and `EXAMPLE` section covers this feature.  
Are we planning to release it with 3.3.0? Or Do we wait 2 weeks for 4.0.0?

**Comment by alinabuzachis** [comment]
@markuman @jatorcasso I guess we can include this if we merge by tomorrow.

File: plugins/inventory/aws_ec2.py [hunk 1]
```diff
@@ -39,11 +39,31 @@
...
+                    - If I(prefix) is specified, final hostname will be I(prefix) +  I(separator) + I(name).
+                type: str
+                default: ''
+                required: False
+            separator:
+                description:
+                    - Value to separate I(prefix) and I(name) when I(prefix) is specified.
+                type: str
+                default: '_'
+                required: False
```
[the remark this answers is not in the document]
Comment (jatorcasso) [reply plugins/inventory/aws_ec2.py:66]: @markuman do we need to add version_added here even though these options already existed? (just were not documented)

[the remark this answers is not in the document]
Comment (markuman) [reply plugins/inventory/aws_ec2.py:66]: Ah yeah, you're right.

**Review by jillr — APPROVED** [review]

**Pull request merged**